### PR TITLE
[posix] remove constants defined by macros

### PR DESCRIPTION
**Pull request by superwhd**

### CODE REVIEW

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -711,7 +707,7 @@ static void processNetifAddrEvent(otInstance *aInstance, struct nlmsghdr *aNetli
                     error = otIp6SubscribeMulticastAddress(aInstance, &addr);
                 }
 
-                logAddrEvent(kAddAddress, !addr.IsMulticast(), addr6, error);
+                logAddrEvent(/*isAdd=*/true, !addr.IsMulticast(), addr6, error);
```
Comment (simonlingoogle) [inline src/posix/platform/netif.cpp:710]:
```suggestion
                logAddrEvent(/* isAdd */ true, !addr.IsMulticast(), addr6, error);
```

**Review by simonlingoogle — APPROVED** [review]
lgtm :+1: 
with style nit

File: src/posix/platform/netif.cpp [hunk 2]
```diff
@@ -664,6 +660,9 @@ static void logAddrEvent(bool isAdd, bool isUnicast, struct sockaddr_in6 &addr6,
 
 static void processNetifAddrEvent(otInstance *aInstance, struct nlmsghdr *aNetlinkMessage)
 {
+    constexpr bool kAddAddress    = true;
```
Comment (simonlingoogle) [inline src/posix/platform/netif.cpp:663]: I wonder if it's necessary to define these constants? Can we just use `true/false`? 
@superwhd @wgtdkp @bukepo

Comment (wgtdkp) [reply src/posix/platform/netif.cpp:663]: SGTM.

**Review by Irving-cl — APPROVED** [review]
LGTM :+1:

**Review by wgtdkp — APPROVED** [review]
LGTM 👍 

May reword the PR title.

**Review by jwhui — APPROVED** [review]

**Pull request merged**